src/java/htsjdk/samtools/CRAMFileWriter.java

```diff
@@ -72,22 +72,69 @@
     private CRAMIndexer indexer;
     private long offset;
 
+    /**
+     * Create a CRAMFileWriter on an output stream. Requires input records to be presorted to match the
+     * sort order defined by the input header.
```

Comment:
could you refer to `{@code samFileHeader}` to make it clear what it's matched against